programs/benchmark/Benchmark.cpp

```diff
@@ -342,14 +346,34 @@ class Benchmark : public Poco::Util::Application
             }
         }
 
-        pool.wait();
-        total_watch.stop();
+        // Setup another signal handler to capture further SIGINT
+        // if the user wants to quit immediately.
+        signal(SIGINT, signalHandler);
+
+        /// run the interrupt handler thread in background detached and
+        /// block if and until signalHandler notifies and then exit.
+        auto interrupt_handler = std::thread{[&]() {
+            while (!cancel_benchmark)
+            {
+            }
+            std::cout << "Second SIGINT received. Quitting immediately."
+                      << "\n";
+            _exit(1);
+        }};
+        interrupt_handler.detach();
+
+        // run the clean up in background detached so it's non-blocking
+        auto cleanup = std::thread{[&]() {
```

Comment:
I had to move the cleanup to run in background & detached in order for the signalHandler to handle the second SIGINT  as it looks like  `pool.wait()` is blocking. However I'm noticing that with just the first Ctrl + C the shutdown is proper sometimes and often times it exits with abort core dumped. I'm not sure if there is a better way to do this.